### DISCUSSION TITLE
C# Support?

### DISCUSSION THREAD

**Discussion by abmarnie** [ramokz/phantom-camera#174] [Question / Help Wanted]
Does this plugin support C#?

More specifically, is it possible for me to use all of the features of this plugin while only writing C# (not writing any GDScript code)?

Looking through the codebase and docs, there does not seem to be any indication of C# support at the moment. If I am right about that, are there any plans to do so?

EDIT: I just remembered that I can just [directly call the GDScript from C#](https://docs.godotengine.org/en/stable/tutorials/scripting/cross_language_scripting.html). So, maybe the answer is true by technicality, even if a little ugly.

**Comment by ramokz** [comment]
Yes, as long as you can grab a reference to a `PCam` node, you should be able to call whatever method you want on it in C#, just like in GDScript. Like you said, the method isn't particularly pretty and feels kinda unsafe, given methods has to be referenced via strings.

Don't think there's another way around that in the current version of Godot, beyond rewriting the addon as a C++ module or, maybe(?), as a GDExtension.

But am not considering doing anything like that until the addon has reached a more mature state.
